redhat-cop/monitoring
# add group for adding new targets

### What does this PR do?
Separates playbooks into two directories monitoring-hosts and monitoring-targets.
Setup-prometheus role now creates monitoring-editors, monitoring-edit-user-1 and ssh-keypair in /tmp/ then adds the public key to the monitoring-edit-user-1 authorized_keys.
It also creates /var/prometheus_targets/ directory owned by monitoring-editors group.
Add-target role creates target definitons in  /var/prometheus_targets/ and can be run without root privileges with ansible_user in monitoring-editors group
Add setup-exporters.yml playbook which deploys exporters on the targets based on the group-membership.
Removes the prometheus group from the inventory and adjusts the playbooks accordingly. 

### How should this be tested?
run the setup-prometheus-grafana and check that the /var/prometheus_targets exists and is owned by monitoring-editors group 
then run the add-targets playbook using key  /tmp/monitoring-edit-user-1_key as monitoring-edit-user-1 with ansible_become: false. Then check /var/prometheus_targets/ where directories with target definitions should appear.

Run setup-exporters.yml with hosts in prometheus_targets, prometheus_target_bind or prometheus_target_haproxy. Then check the running docker containers on the host (prometheus exporter should be running there).

### Is there a relevant Issue open for this?
Provide a link to any open issues that describe the problem you are solving.
resolves #<number>

### Other Relevant info, PRs, etc.
Please provide link to other PRs that may be related (blocking, resolves, etc. etc.)

### People to notify
cc: @redhat-cop/monitoring
